webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala

```diff
@@ -103,6 +135,581 @@ object RenderInheritedProperties {
+     * Provider is also merged into the datastructure from the virtual type.
+     * It is transformed from an optional domain.MachineInfo because the type is still serialized in the None case
+     */
+    final case class MachineInfo(
+        id:                             Option[MachineUuid],
+        @jsonField("type") machineType: MachineType,
+        provider:                       Option[VmType],
+        manufacturer:                   Option[Manufacturer],
+        serialNumber:                   Option[String]
+    )
+
+    object MachineInfo {
+      implicit val transformer: Transformer[Option[nodes.MachineInfo], MachineInfo] = {
+        _.map {
+          case nodes.MachineInfo(id, UnknownMachineType, serial, manufacturer)     =>
+            MachineInfo(Some(id), MachineType.UnknownMachineType, None, manufacturer, serial)
+          case nodes.MachineInfo(id, PhysicalMachineType, serial, manufacturer)    =>
+            MachineInfo(Some(id), MachineType.PhysicalMachineType, None, manufacturer, serial)
+          case nodes.MachineInfo(id, VirtualMachineType(vm), serial, manufacturer) =>
+            MachineInfo(Some(id), MachineType.VirtualMachineType, Some(vm), manufacturer, serial)
+        }.getOrElse {
+          MachineInfo(None, MachineType.NoMachine, None, None, None)
+        }
+      }
+    }
+
+    /**
+     * The structure does not directly match the domain object, nodeKind is a contextual value of the node added to each AgentInfo
+     */
+    final case class Management(
+        name:         AgentType,
+        version:      Option[AgentVersion],
+        capabilities: Chunk[String],
+        nodeKind:     NodeKind
+    )
+    object Management {
+      implicit val transformer: Transformer[NodeInfo, Chunk[Management]] = (info: NodeInfo) => {
+        val agents = info.agentsName.map { agent =>
+          val capabilities = agent.capabilities.map(_.value).toList.sorted
+          Management(
+            agent.agentType,
+            agent.version,
+            Chunk.fromIterable(capabilities),
+            info.nodeKind
+          )
+        }
+        Chunk.fromIterable(agents)
+      }
+    }
+
+    /**
+     * The structure does not directly match the domain object, it aggregates some fields
+     */
+    final case class ManagementDetails(
+        cfengineKeys: Chunk[SecurityToken],
+        cfengineUser: String
+    )
+    object ManagementDetails {
+      implicit val transformer: Transformer[NodeInventory, ManagementDetails] = Transformer
+        .define[NodeInventory, ManagementDetails]
+        .withFieldComputed(_.cfengineKeys, n => Chunk.fromIterable(n.agents.map(_.securityToken)))
+        .withFieldComputed(_.cfengineUser, _.main.rootUser)
+        .buildTransformer
+    }
+
+    final case class DisplayDateTime(value: DateTime) extends AnyVal
+
+    /**
+     * Needed because datetime is serialized differently and therefore needs the DisplayDateTime wrapper
```

Comment:
I'm not sure we want that. Having only one date format for everything is more UX friendlier

Reply:
I made the `Bios` keep the same format of datatime for the `releaseDate` field in the API versions before v20 https://github.com/Normation/rudder/pull/5656/commits/b591687a7572efef195c57d3d289553551157d4c